core/src/main/java/org/elasticsearch/discovery/Discovery.java

```diff
@@ -87,4 +88,10 @@ public FailedToCommitClusterStateException(String msg, Throwable cause, Object..
             super(msg, cause, args);
         }
     }
+
+    /**
+     * @return stats about the discovery
+     */
+    PendingClusterStateStats stats();
```

Comment:
OK. I see - valid point. If you're up to it we can implement a whole registry based on NamedWriteableRegistry and NamedWriteable. Alternatively I'm good with hard coding DiscoveryStats to include whatever ZenDiscovery needs (pending cluster state queue stats only now). We can open it up later if we need to.